src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -25,8 +25,15 @@ internal RuntimeTypeHandle GetNativeHandle() =>
         internal RuntimeType GetTypeChecked() =>
             m_type ?? throw new ArgumentNullException(null, SR.Arg_InvalidHandle);
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        internal static extern bool IsInstanceOfType(RuntimeType type, [NotNullWhen(true)] object? o);
+        internal static bool IsInstanceOfType(RuntimeType type, [NotNullWhen(true)] object? o)
+        {
+            if (type is null)
+            {
+                throw new ArgumentNullException(SR.Arg_InvalidHandle);
+            }
+
+            return CastHelpers.IsInstanceOfAny(type.GetUnderlyingNativeHandle().ToPointer(), o) is not null;
```

Comment:
```suggestion
            bool ret = CastHelpers.IsInstanceOfAny(type.GetUnderlyingNativeHandle().ToPointer(), o) is not null;
            GC.KeepAlive(type);
            return ret;
```
I think this needs KeepAlive.

Reply:
Shoot, thanks. Missed this one when applying feedback.